front/app/modules/commercial/insights/hooks/useInsightsInputs.ts

```diff
@@ -1,25 +1,57 @@
 import { useState, useEffect } from 'react';
+import { getPageNumberFromUrl } from 'utils/paginationUtils';
 import {
   insightsInputsStream,
   IInsightsInputData,
 } from '../services/insightsInputs';
 
-const useInsightsViews = (viewId: string) => {
-  const [insightsViews, setInsightsViews] = useState<
+const defaultPageSize = 20;
+
+interface Options {
+  pageSize: number;
+  pageNumber: number;
+}
+
+export interface IUseInpightsInputsOutput {
+  list: IInsightsInputData[] | undefined | null;
+  loading: boolean;
+  onChangePage: (pageNumber: number) => void;
+  currentPage: number;
+}
+
+const useInsightsInputs = (viewId: string, options?: Partial<Options>) => {
+  const [insightsInputs, setInsightsInputs] = useState<
     IInsightsInputData[] | undefined | null | Error
   >(undefined);
 
+  const [currentPage, setCurrentPage] = useState<number | null>(
+    options?.pageNumber || null
+  );
+  const [lastPage, setLastPage] = useState<number | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+
   useEffect(() => {
-    const subscription = insightsInputsStream(viewId).observable.subscribe(
-      (insightsViews) => {
-        setInsightsViews(insightsViews.data);
-      }
-    );
+    setLoading(true);
+    const subscription = insightsInputsStream(viewId, {
+      queryParameters: {
+        'page[number]': options?.pageNumber || 1,
+        'page[size]': options?.pageSize || defaultPageSize,
+      },
+    }).observable.subscribe((insightsInputs) => {
+      setInsightsInputs(insightsInputs.data);
+      setCurrentPage(getPageNumberFromUrl(insightsInputs.links?.self));
```

Comment:
Well, that way if the request takes a long time, the page doesn't change before the content, but it might be a bit overkill, and we have to loading prop too so removing for simplicity indeed. (It's also kind of a leftover from the class component version of this)